packages/core/workflows-sdk/src/utils/composer/type.ts

```diff
@@ -73,18 +85,20 @@ export type WorkflowData<T = unknown> = (T extends Array<infer Item>
   }
 
 export type CreateWorkflowComposerContext = {
-  hooks_: string[]
-  hooksCallback_: Record<string, Function[]>
+  __type: string
+  hooks_: {
+    declared: string[]
+    registered: string[]
+  }
+  hooksCallback_: Record<string, any>
   workflowId: string
   flow: OrchestratorBuilder
   handlers: WorkflowHandler
   stepBinder: <TOutput = unknown>(
     fn: StepFunctionResult
   ) => WorkflowData<TOutput>
-  hookBinder: <TOutput = unknown>(
-    name: string,
-    fn: Function
-  ) => WorkflowData<TOutput>
+  // TODO change the typings
+  hookBinder: (name: string, fn: Function) => any
```

Comment:
suggestion: should we update the types here? should return void for example

Reply:
Sure